search.go

```diff
@@ -397,20 +414,11 @@ func (r *SearchRequest) UnmarshalJSON(input []byte) error {
 	r.Score = temp.Score
 	r.SearchAfter = temp.SearchAfter
 	r.SearchBefore = temp.SearchBefore
-	r.Query, err = query.ParseQuery(temp.Q)
-	if err != nil {
+	if r.Query, err = query.ParseQuery(temp.Q); err != nil {
 		return err
 	}
 
-	if r.Size < 0 {
```

Comment:
I believe its in our interest to keep these checks after the query parsing to override any negative integers from the user.